src/Datadog.Trace/Span.cs

```diff
@@ -60,7 +60,11 @@ internal Span(IDatadogTracer tracer, SpanContext parent, string operationName, s
         /// <summary>
         /// Gets the service name
         /// </summary>
-        public string ServiceName => _context.ServiceName;
+        public string ServiceName
+        {
+            get { return _context.ServiceName; }
+            internal set { _context.ServiceName = value; }
```

Comment:
I'm late to the party sorry, but why not let the non opentracing tracer change the service name too?

Reply:
You're right, there's nothing stopping us from allowing that as well. We just need to drop `internal` here. I was trying to fix the blocking issue only without changing anything else.